shrine_with_cart/mdc_100_series/lib/short_bottom_sheet.dart

```diff
@@ -0,0 +1,578 @@
+    assert(nullOk != null);
+    assert(context != null);
+    final _ShortBottomSheetState result = context
+        .ancestorStateOfType(const TypeMatcher<_ShortBottomSheetState>());
+    if (nullOk || result != null) {
+      return result;
+    }
+    throw FlutterError(
+        'ShortBottomSheet.of() called with a context that does not contain a ShortBottomSheet.\n');
+  }
+}
+
+class _ShortBottomSheetState extends State<ShortBottomSheet>
+    with TickerProviderStateMixin {
+  final GlobalKey _shortBottomSheetKey =
+      GlobalKey(debugLabel: 'Short bottom sheet');
+  // The padding between the left edge of the Material and the shopping cart icon
+  double _cartPadding;
+  // The width of the Material, calculated by _getWidth & based on the number of
+  // products in the cart.
+  double _width;
+  // Controller for the opening and closing of the ShortBottomSheet
+  AnimationController _controller;
+  // Animations for the opening and closing of the ShortBottomSheet
+  Animation<double> _widthAnimation;
+  Animation<double> _heightAnimation;
+  Animation<double> _thumbnailOpacityAnimation;
+  Animation<double> _cartOpacityAnimation;
+  Animation<double> _shapeAnimation;
+  Animation<Offset> _slideAnimation;
+  // Curves that represent the two curves that compose the emphasized easing curve.
+  final Cubic _accelerateCurve = const Cubic(0.548, 0.0, 0.757, 0.464);
+  final Cubic _decelerateCurve = const Cubic(0.23, 0.94, 0.41, 1.0);
+  final double _cartHeight = 56.0;
+
+  @override
+  void initState() {
+    super.initState();
+    _adjustCartPadding(0);
+    _updateWidth(0);
+    _controller = AnimationController(
+        duration: const Duration(milliseconds: 300), vsync: this);
+    _slideAnimation = TweenSequence(
+      <TweenSequenceItem<Offset>>[
+        TweenSequenceItem<Offset>(
+            tween: Tween<Offset>(
+              begin: Offset(1.0, 0.0),
+              end: Offset(0.6, 0.0),
+            ).chain(CurveTween(curve: _accelerateCurve)),
+            weight: 1.0 / 6.0),
+        TweenSequenceItem<Offset>(
+            tween: Tween<Offset>(
+              begin: Offset(0.6, 0.0),
+              end: Offset(0.0, 0.0),
+            ).chain(CurveTween(curve: _decelerateCurve)),
+            weight: 5.0 / 6.0),
+      ],
+    ).animate(
+      CurvedAnimation(parent: widget.hideController, curve: Interval(0.0, 1.0)),
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  // Updates the animations for the opening/closing of the ShortBottomSheet,
+  // using the size of the screen.
+  void _updateAnimations(BuildContext context) {
+    Size screenSize = MediaQuery.of(context).size;
+    double mediaWidth = screenSize.width;
+    double mediaHeight = screenSize.height;
+    double cornerRadius = 24.0;
+
+    _widthAnimation = TweenSequence(
+      <TweenSequenceItem<double>>[
+        TweenSequenceItem<double>(
+          // 1/6 of duration = 40% (0.4) of property delta
```

Comment:
I feel like a number or something is missing from that math comment.

Reply:
Removed the 0.4 for clarity, but this function will be changing once curve issues are sorted out.